docs/changes/newsfragments/5381.new

```diff
@@ -0,0 +1,2 @@
+Add a register_name keyword to the ParameterBase constructor. If supplied, this value
+will be used to register parameters in datasets instead of the default full_name.
```

Comment:
```suggestion
will be used to register parameters in datasets instead of the default full_name.
Note that the new register_name keyword argument does not work for MultiParameter and ArrayParameter.
```